src/test/kotlin/com/kafkakotlin/demo/users/UserControllerTest.kt

```diff
@@ -0,0 +1,32 @@
+package com.kafkakotlin.demo.users
+
+import io.mockk.every
+import io.mockk.just
+import io.mockk.mockk
+import io.mockk.runs
+import io.mockk.slot
+import org.junit.jupiter.api.Assertions.assertEquals
+import org.junit.jupiter.api.DisplayName
+import org.junit.jupiter.api.Test
+
+@DisplayName("User Controller Tests")
+internal class UserControllerTest {
+    private val userService = mockk<UserService>()
+    private val underTest = UserController(userService = userService)
+
+    private val dummyUser = User(
+        username = "Steve",
+        email = "steve@steve.com",
+        password = "super-strong-password"
+    )
+
+    @Test
+    fun `should call the UserService with the correct parameters`() {
+        val userSlot = slot<User>()
+        every { userService.createUser(capture(userSlot)) } just runs
+
+        userService.createUser(dummyUser)
```

Comment:
Looks like you are testing mockk here. Shouldn't this be calling your Controller

Reply:
Using all that unusual `slot` stuff should be a bit of a red flag

Reply:
is `slot` not what we should be using for testing what's passed through as a parameter? Or just generally we shouldnt be doing that kind of test?

Reply:
Ah, just seen your comment below on `verify`

Reply:
`slot` is fine, but its generally used for really complex stuff where you want to assert on some updates that have been made to your `user`

`verify` is perfect for your scenario though

Reply:
👍 cheers, just tried it now and looks good. Hadn't realised I called `userService` rather than `underTest` either.